api/queries_actions.go

```diff
@@ -0,0 +1,33 @@
+package api
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/cli/cli/internal/ghinstance"
+	"github.com/cli/cli/internal/ghrepo"
+)
+
+func (c Client) JobLog(repo ghrepo.Interface, jobID string) (io.ReadCloser, error) {
```

Comment:
I need access to `c.http.Do`; is there a way to move this and still have that access?

Reply:
I would move away from `api.Client`, since that object has never been a useful abstraction, and just receive an `*http.Client` argument instead.

Reply:
ah. I thought `api.Client` carried some useful state with it but I see that it doesn't and just provides methods.

I can clean this up, but should it be our explicit intention to remove `api.Client` entirely in favor of functions that accept `*http.Client`? It's confusing to have both modes.

Reply:
I've made the change but leaving this unresolved because i'm interested in the wider discussion.

Reply:
It's my explicit intention to remove `api.Client` since that's an abstraction that was ported from hub, but in the version that it ended up as in our codebase it provides no value except as a vessel for a reference to a `*http.Client` internally and for exposing methods such as GraphQL and REST. Either of these methods could just as easily be passed a `*http.Client*` as an argument.